### PR TITLE
Improve ccache use on github workflows

### PR DESCRIPTION
**Pull request by sethrj**
This PR:
- fixes ccache invalidation due to different runner host names
- increases ccache size for windows which I think is causing issues with invalidation
- adds a cron job so that every sunday morning all ccache entries are cleared and new ones regenerated
- Updates the `pr` filename for consistency with the workflow trigger
- Update the `merge_group` workflow so we can try it again

### CODE REVIEW

File: .github/workflows/clear-cache.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: clear-cache
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  clear-cache:
+    runs-on: ubuntu-latest
```
Comment (pcanal) [inline .github/workflows/clear-cache.yml:9]: Should it run on all the platform with CCache? (I.e. eg Windows?)

Comment (sethrj) [reply .github/workflows/clear-cache.yml:9]: Nope. All it's doing is making GitHub calls; the cache itself is platform agnostic.

**Comment by sethrj** [comment]
Ping @pcanal

**Comment by sethrj** [comment]
Ping @pcanal

**Pull request merged**